src/mozilla_sec_eia/models/sec10k/sec_output_table.py

```diff
@@ -0,0 +1,367 @@
+    ).rename(columns={"central_index_key": "subsidiary_cik"})
+    # if a subsidiary doesn't have a CIK and has a null location
+    # but its company name was assigned a CIK (with a different location)
+    # then assign that CIK to the subsidiary
+    ex21_with_cik_df = ex21_with_cik_df.merge(
+        closest_match_df[["company_name", "central_index_key"]],
+        how="left",
+        on="company_name",
+    ).rename(columns={"central_index_key": "company_name_merge_cik"})
+    ex21_with_cik_df["subsidiary_cik"] = ex21_with_cik_df["subsidiary_cik"].where(
+        ~(ex21_with_cik_df.subsidiary_cik.isnull())
+        | ~(ex21_with_cik_df.location_of_inc.isnull()),
+        ex21_with_cik_df["company_name_merge_cik"],
+    )
+    ex21_with_cik_df = ex21_with_cik_df.drop(columns="company_name_merge_cik")
+    ex21_with_cik_df = ex21_with_cik_df.rename(
+        columns={"subsidiary_cik": "central_index_key"}
+    )
+    return ex21_with_cik_df
+
+
+def create_sec_company_id_for_ex21_subs(ex21_df: pd.DataFrame) -> pd.DataFrame:
+    """Create an sec_company_id for Ex. 21 subsidiaries.
+
+    This is a unique identifier string for Ex. 21 subsidiaries.
+    This ID is necessary for tracking subsidiaries who aren't ultimately
+    matched to a 10K filer company.
+    """
+    ex21_df.loc[:, "sec_company_id"] = (
+        ex21_df["parent_company_cik"]
+        + "_"
+        + ex21_df["company_name"]
+        + "_"
+        + ex21_df["location_of_inc"]
+    )
+    return ex21_df
+
+
+@multi_asset(
+    ins={
+        "ex21_df": AssetIn("ex21_company_ownership_info"),
+    },
+    outs={
+        "clean_ex21_subsidiary_table": AssetOut(
+            io_manager_key="pandas_parquet_io_manager",
+        )
+    },
+    partitions_def=year_quarter_partitions,
+)
+def clean_ex21_table(
+    ex21_df: pd.DataFrame, sec10k_filing_metadata: pd.DataFrame
+) -> pd.DataFrame:
+    """Clean Ex. 21 table of subsidiaries before combining with basic 10k table."""
+    ex21_df.loc[:, "filename"] = convert_ex21_id_to_filename(ex21_df)
+    ex21_df = ex21_df.drop(columns=["id"])
+    ex21_df = _add_report_year_to_sec(ex21_df, sec10k_filing_metadata)
+    ex21_df = ex21_df.rename(
+        columns={"subsidiary": "company_name", "loc": "location_of_inc"}
+    )
+    ex21_df = clean_location_of_inc(ex21_df)
+    ex21_df = clean_company_name(ex21_df)
+    ex21_df = add_parent_company_cik(ex21_df, sec10k_filing_metadata)
+    # add an sec_company_id, ultimately this ID become the subsidiary's CIK
+    # if the subsidiary is matched to an SEC filer
+    ex21_df = create_sec_company_id_for_ex21_subs(ex21_df=ex21_df)
+    ex21_df = _flatten_sec_companies_across_time(ex21_df)
+
+    return ex21_df
+
+
+@multi_asset(
+    ins={
+        "basic_10k_df": AssetIn("basic_10k_company_info"),
+        "clean_ex21_df": AssetIn("clean_ex21_subsidiary_table"),
+        # specify an io_manager_key?
+    },
+    outs={
+        "out_sec_10k__parents_and_subsidiaries": AssetOut(
```

Comment:
I guess this isn't actually the final output table, but more like an intermediate asset. We still need to conduct record linkage on it and add in a column with the connecting `utility_id_eia`.